src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1405,6 +1428,141 @@ private void WritePackageDependencies()
                 }
             }
 
+            // This implementation was taken from PreprocessPackageDependenciesDesignTime.ExecuteCore
+            // with some minor modifications to use WriteItem instead of saving it into an array.
+            private void WritePackageDependenciesDesignTime()
+            {
+                List<string> packageDependencies = GetPackageDependencies();
+
+                var implicitPackageReferences = GetImplicitPackageReferences(_task.DefaultImplicitPackages);
+
+                // We have two types of data:
+                //
+                // 1) "PackageDependencies" which place a package in a given target/hierarchy
+                // 2) "PackageDefinitions" which provide general metadata about a package
+                //
+                // First, we scan PackageDependencies to build the set of packages in our target.
+
+                var allowItemSpecs = packageDependencies.ToHashSet(StringComparer.OrdinalIgnoreCase);
+
+                // Second, find PackageDefinitions that match our allowed item specs
+
+                foreach (var package in _lockFile.Libraries)
+                {
+                    if (string.IsNullOrEmpty(package.Name) || !allowItemSpecs.Contains(package.Name))
```

Comment:
The `allowedItemSpecs` are of the form `{packageName}/{packageVersion}` (from the `GetPackageId` method).  So `package.Name` will never match this.

I missed when this was introduced when the `PackageDefinitionItem` type was added.

Are you doing any ad-hoc testing of these changes?  I think this means that no packages would be displayed as dependencies in Solution Explorer.

Reply:
fixed